### ISSUE TITLE
can you share the demo code for 3D box visualization

### ISSUE THREAD

**Issue by jkstyle2** [V-DETR/V-DETR#6]
Thanks for sharing your great work!
I'd like to qualitatively check the results with the previous papers.
Can you share how to simply visualize the outputs of the model like ply or pcd format?

Also, what I got from the test script in README is as follows.
The mAPs are 70.35, 57.56 for 0.25, 0.5 iou, which are quite different from your results.
Can you explain why there exists the differences?

> Test model; Metrics mAP0.25, mAP0.50: 70.35, 57.56
> AR0.25, AR0.50: 92.72, 79.35

> IOU Thresh=0.25
> cabinet Average Precision: 52.63
> bed Average Precision: 90.51
> chair Average Precision: 94.62
> sofa Average Precision: 90.42
> table Average Precision: 77.06
> door Average Precision: 60.30
> window Average Precision: 52.43
> bookshelf Average Precision: 62.71
> picture Average Precision: 20.76
> counter Average Precision: 70.51
> desk Average Precision: 78.33
> curtain Average Precision: 70.58
> refrigerator Average Precision: 58.24
> showercurtrain Average Precision: 64.69
> toilet Average Precision: 99.54
> sink Average Precision: 74.58
> bathtub Average Precision: 89.45
> garbagebin Average Precision: 58.92
> cabinet Recall: 93.55
> bed Recall: 96.30
> chair Recall: 98.54
> sofa Recall: 98.97
> table Recall: 96.86
> door Recall: 90.58
> window Recall: 91.84
> bookshelf Recall: 97.40
> picture Recall: 56.31
> counter Recall: 90.38
> desk Recall: 98.43
> curtain Recall: 95.52
> refrigerator Recall: 100.00
> showercurtrain Recall: 96.43
> toilet Recall: 100.00
> sink Recall: 81.63
> bathtub Recall: 93.55
> garbagebin Recall: 92.64

> IOU Thresh=0.5
> cabinet Average Precision: 35.79
> bed Average Precision: 81.63
> chair Average Precision: 88.50
> sofa Average Precision: 84.66
> table Average Precision: 69.24
> door Average Precision: 45.07
> window Average Precision: 32.45
> bookshelf Average Precision: 49.08
> picture Average Precision: 14.52
> counter Average Precision: 54.06
> desk Average Precision: 54.80
> curtain Average Precision: 50.71
> refrigerator Average Precision: 44.28
> showercurtrain Average Precision: 44.33
> toilet Average Precision: 94.75
> sink Average Precision: 55.05
> bathtub Average Precision: 84.93
> garbagebin Average Precision: 52.18
> cabinet Recall: 68.82
> bed Recall: 92.59
> chair Recall: 92.69
> sofa Recall: 96.91
> table Recall: 88.00
> door Recall: 66.38
> window Recall: 62.06
> bookshelf Recall: 88.31
> picture Recall: 33.33
> counter Recall: 71.15
> desk Recall: 89.76
> curtain Recall: 76.12
> refrigerator Recall: 87.72
> showercurtrain Recall: 82.14
> toilet Recall: 96.55
> sink Recall: 69.39
> bathtub Recall: 87.10

**Comment by yichaoshen-MS** [comment]
Hi, Did you prepare the data follow README? Could you share full command and outputs during inferencing?

**Comment by jkstyle2** [comment]
yes, i followed the preparation part for scannet. Here is the output result below.


```
V-DETR$ python main.py --dataset_name scannet --dataset_root_dir /mnt/synology/opendataset/scannet/processed/scannet/scannet_train_detection_data/ --meta_data_dir /mnt/synology/opendataset/scannet/processed/scannet/meta_data/ --test_only --test_ckpt ckpts/scannet_540ep.pth --auto_test
/home/sven/miniconda3/envs/openmmlab/lib/python3.8/site-packages/MinkowskiEngine/__init__.py:36: UserWarning: The environment variable `OMP_NUM_THREADS` not set. MinkowskiEngine will automatically set `OMP_NUM_THREADS=16`. If you want to set `OMP_NUM_THREADS` manually, please export it on the command line before running a python script. e.g. `export OMP_NUM_THREADS=12; python your_program.py`. It is recommended to set it below 24.
  warnings.warn(
Called with args: Namespace(angle_conf=False, angle_nms=False, angle_type='', auto_test=True, axis_align_test=False, base_lr=0.0007, batchsize_per_gpu=1, checkpoint_dir=None, clip_gradient=0.1, cls_loss='focalloss_0.25', color_contrastp=0.0, color_drop=0.0, color_jitterp=0.0, color_mean=-1, coloraug_sunrgbd=False, conf_thresh=0.0, dataset_name='scannet', dataset_num_workers=8, dataset_root_dir='/mnt/synology/opendataset/scannet/processed/scannet/scannet_train_detection_data/', dec_dim=256, dec_dropout=0.1, dec_ffn_dim=256, dec_nhead=4, dec_nlayers=9, depth=34, dist_url='tcp://localhost:12345', empty_pt_thre=5, enc_dim=256, eval_every_epoch=10, filt_empty=True, filter_biases_wd=False, final_lr=1e-06, hard_anchor=False, hue_sat='0.5_0.2_0.0', inplanes=64, iou_type='giou', is_bilable=True, layer_idx=0, log_every=10, log_metrics_every=20, log_scale=512, loss_angle_cls_weight=0.1, loss_angle_reg_weight=0.5, loss_center_weight=1, loss_giou_weight=2, loss_no_object_weight=0, loss_sem_cls_weight=3, loss_size_weight=0.5, lr_scheduler='cosine', matcher_anglecls_cost=0, matcher_anglereg_cost=0, matcher_center_cost=1, matcher_cls_cost=3, matcher_giou_cost=2, matcher_objectness_cost=0, matcher_size_cost=0.5, max_epoch=540, meta_data_dir='/mnt/synology/opendataset/scannet/processed/scannet/meta_data/', mink_syncbn=True, minkowski=True, mlp_act='relu', mlp_dropout=0.3, mlp_norm='bn1d', mlp_sep=True, model_name='vdetr', ngpus=1, nms_iou=0.25, no_3d_nms=False, no_cls_nms=False, no_first_repeat=True, no_mink_first_pool=True, no_per_class_proposal=False, normal_trans=False, nqueries=1024, nsemcls=-1, num_points=100000, num_stages=4, point_cls_loss_weight=0.05, pos_for_key=False, preenc_npoints=4096, proj_nohid=True, q_content='random', querypos_mlp=True, repeat_num=5, rot_ratio=5.0, rotated_nms=False, rpe_dim=128, rpe_quant='bilinear_4_10', save_separate_checkpoint_every_epoch=1, scale_ratio=0.4, seed=0, share_selfattn=False, start_epoch=-1, stem_bn=True, step_epoch='', test_ckpt='ckpts/scannet_540ep.pth', test_no_nms=False, test_only=True, test_size=False, trans_ratio=0.4, use_cls_confidence_only=False, use_color=False, use_fpn=True, use_normals=False, use_old_type_nms=False, use_superpoint=False, voxel_size=0.01, wandb_activate=True, wandb_entity=None, wandb_key='', wandb_project='vdetr', warm_lr=1e-06, warm_lr_epochs=9, weight_decay=0.1, woexpand_conv=True, xyz_color=False)
hard_anchor False True
use_color False True
xyz_color False True
start_epoch -1 0
save_separate_checkpoint_every_epoch 1 -1
/mnt/synology/opendataset/scannet/processed/scannet/meta_data/scannetv2_val.txt
kept 312 scans out of 312
val 312 scans before filter
val 312 scans after filter
random init backbone
/home/sven/miniconda3/envs/openmmlab/lib/python3.8/site-packages/torch/functional.py:507: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at /opt/conda/conda-bld/pytorch_1704987301168/work/aten/src/ATen/native/TensorShape.cpp:3549.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
random init decoder
/home/sven/miniconda3/envs/openmmlab/lib/python3.8/site-packages/torch/nn/functional.py:4316: UserWarning: Default grid_sample and affine_grid behavior has changed to align_corners=False since 1.3.0. Please specify align_corners=True if the old behavior is desired. See the documentation of grid_sample for details.
  warnings.warn(
Evaluate ; Batch [0/312]
Evaluate ; Batch [10/312]
Evaluate ; Batch [20/312]
Evaluate ; Batch [30/312]
Evaluate ; Batch [40/312]
Evaluate ; Batch [50/312]
Evaluate ; Batch [60/312]
Evaluate ; Batch [70/312]
Evaluate ; Batch [80/312]
Evaluate ; Batch [90/312]
Evaluate ; Batch [100/312]
Evaluate ; Batch [110/312]
Evaluate ; Batch [120/312]
Evaluate ; Batch [130/312]
Evaluate ; Batch [140/312]
Evaluate ; Batch [150/312]
Evaluate ; Batch [160/312]
Evaluate ; Batch [170/312]
Evaluate ; Batch [180/312]
Evaluate ; Batch [190/312]
Evaluate ; Batch [200/312]
Evaluate ; Batch [210/312]
Evaluate ; Batch [220/312]
Evaluate ; Batch [230/312]
Evaluate ; Batch [240/312]
Evaluate ; Batch [250/312]
Evaluate ; Batch [260/312]
Evaluate ; Batch [270/312]
Evaluate ; Batch [280/312]
Evaluate ; Batch [290/312]
Evaluate ; Batch [300/312]
Evaluate ; Batch [310/312]
====================
Test model; Metrics mAP0.25, mAP0.50: 70.35, 57.56
AR0.25, AR0.50: 92.72, 79.35
-----
IOU Thresh=0.25
cabinet Average Precision: 52.63
bed Average Precision: 90.51
chair Average Precision: 94.62
sofa Average Precision: 90.42
table Average Precision: 77.06
door Average Precision: 60.30
window Average Precision: 52.43
bookshelf Average Precision: 62.71
picture Average Precision: 20.76
counter Average Precision: 70.51
desk Average Precision: 78.33
curtain Average Precision: 70.58
refrigerator Average Precision: 58.24
showercurtrain Average Precision: 64.69
toilet Average Precision: 99.54
sink Average Precision: 74.58
bathtub Average Precision: 89.45
garbagebin Average Precision: 58.92
cabinet Recall: 93.55
bed Recall: 96.30
chair Recall: 98.54
sofa Recall: 98.97
table Recall: 96.86
door Recall: 90.58
window Recall: 91.84
bookshelf Recall: 97.40
picture Recall: 56.31
counter Recall: 90.38
desk Recall: 98.43
curtain Recall: 95.52
refrigerator Recall: 100.00
showercurtrain Recall: 96.43
toilet Recall: 100.00
sink Recall: 81.63
bathtub Recall: 93.55
garbagebin Recall: 92.64
-----
IOU Thresh=0.5
cabinet Average Precision: 35.79
bed Average Precision: 81.63
chair Average Precision: 88.50
sofa Average Precision: 84.66
table Average Precision: 69.24
door Average Precision: 45.07
window Average Precision: 32.45
bookshelf Average Precision: 49.08
picture Average Precision: 14.52
counter Average Precision: 54.06
desk Average Precision: 54.80
curtain Average Precision: 50.71
refrigerator Average Precision: 44.28
showercurtrain Average Precision: 44.33
toilet Average Precision: 94.75
sink Average Precision: 55.05
bathtub Average Precision: 84.93
garbagebin Average Precision: 52.18
cabinet Recall: 68.82
bed Recall: 92.59
chair Recall: 92.69
sofa Recall: 96.91
table Recall: 88.00
door Recall: 66.38
window Recall: 62.06
bookshelf Recall: 88.31
picture Recall: 33.33
counter Recall: 71.15
desk Recall: 89.76
curtain Recall: 76.12
refrigerator Recall: 87.72
showercurtrain Recall: 82.14
toilet Recall: 96.55
sink Recall: 69.39
bathtub Recall: 87.10
```

**Comment by yichaoshen-MS** [comment]
Hi, I think the issue can be solved in [V-DETR/V-DETR#8](https://github.com/V-DETR/V-DETR/issues/8), due to the difference way to process dataset.
